config/searchConfig.js

```diff
@@ -0,0 +1,39 @@
+const fields = ["name", "username", "tags", "location.name"];
+const defaultPath = "name"
+function getQueryString(queryArr) {
+  const queryStrings = queryArr.map((term) => {
+    const orConditions = fields.map((field) => `${field}:${term}`).join(" OR ");
+    return `(${orConditions})`;
+  });
+
+  return queryStrings.join(" AND ");
+}
+export function generateAggreation(termsArray) {
+  const query = getQueryString(termsArray);
+  const agg = [
+    {
+      $search: {
+        compound: {
+          must: [
+            {
+              queryString: {
+                query: query,
+                defaultPath: defaultPath,
+              },
+            },
+          ],
+        },
+      },
+    },
+    {
+      $project: {
+        name: 1,
+        username: 1,
+        "location.name": 1,
+        tags: 1,
+        score: { $meta: "searchScore" },
+      },
+    },
+  ];
+  return agg;
+}
```

Comment:
Is this used in multiple places? I am not sure it needs to be in the config, it looks like code to me

Reply:
No it is used in one place only. But since it is search query putting it with the search function can make it a bit messy. Since it is query string which is also a somekind of config , so I thought to put all these functions in a separate file searchConfig.js

Reply:
Sorry for my slow reply. I think this is more confusing as people would not look there for it. Best to keep it in the actual place for now and it can be refactored/optimised later if required

Reply:
ok I am implementing the changes required

Reply:
Thank you 👍